modules/apps/commerce/commerce-order-content-web/src/main/resources/META-INF/resources/js/info_box/OrderNotesView.js

```diff
@@ -0,0 +1,171 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import ClayButton from '@clayui/button';
+import {useModal} from '@clayui/modal';
+import {CommerceServiceProvider} from 'commerce-frontend-js';
+import {openConfirmModal, openToast, sub} from 'frontend-js-web';
+import moment from 'moment';
+import React, {useCallback, useEffect, useState} from 'react';
+
+import InfoBoxModalNotes from '../InfoBoxModalNotes';
+
+const OrderNotesView = ({
+	buttonDisplayType,
+	elementId,
+	hasPermission,
+	label,
+	namespace,
+	orderId,
+	spritemap,
+}) => {
+	const {observer, onOpenChange, open} = useModal();
+	const [inputValue, setInputValue] = useState();
+	const [isRestricted, setIsRestricted] = useState(false);
+	const [isValid, setIsValid] = useState(true);
+	const [notes, setNotes] = useState([]);
+	const [updateNotes, setUpdateNotes] = useState(notes);
```

Comment:
For now it could stay, conceptually it's fine. This opens to the fact that ideally the modal should have always stayed open when posting / reading comments.